### PR TITLE
Contracts tab

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: doc-content/enterprise-only/employee-rostering/optashift-ER-overview-source-con.adoc [hunk 1]
```diff
@@ -11,7 +11,7 @@ While some code is shared between the client and the server, you can build and u
 In addition to the two main components, the employee rostering template contains a generator of random source data (useful for demonstration and testing purposes) and a benchmarking application.
 
 .Modules and key classes
-The Java source code of the employee rostering template contains several Maven modules. Each of these modules includes a separate aven project file (`pom.xml`), but they are intended for building in a common project. 
+The Java source code of the employee rostering template contains several Maven modules. Each of these modules includes a separate aven project file (`pom.xml`), but they are intended for building in a common project.
```
Comment (Christopher-Chianelli) [inline doc-content/enterprise-only/employee-rostering/optashift-ER-overview-source-con.adoc]: type: `aven` should be `maven`

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
+[id='er-contracts-proc']
+= Entering the list of contracts
+
+You must enter the list of all types of contracts that the business uses for employees.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: all of the types

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+A contract determines the maximum time that the employee can work in a day, calendar week, calendar month, or calendar year.
+
+You can set any of the limitations or none at all is a contract. For example, a part-time employee might not be allowed to work more than 20 hours per week, while a full-time employee might be limited to 10 hours in a day and 1800 hours during a year. Another contract might include no limitations on worked hours.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: Rethink. I think this is missing something or has something extra:
You can set any of the limitations or none at all is a contract.

For readability and esp for translations, consider using the same term or phrase here:
more than 20 hours in a week, while a full-time employee might be limited to 10 hours in a day and 1800 hours in a year.  Your way is more interesting to read, but...

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+You must enter all work time limits for contracts in minutes.
+
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: Any prereqs? Do I do this on paper :-) Something like:
Prerequisites
You have an employee rostering  application that you want to update.

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
++
+You can see the numbers of currently visible contracts in the top right part of the browser window, for example, `1-15 of 34`. You can use the *<* and *>* buttons to display other contracts in the list.
++
+You can enter any part of an contract name in the *Search* box to search for contracts.
++
+. Complete the following steps to add a new contract:
+.. Click *Add Contract*.
+.. Enter the name of the contract into the text field under *Name*.
+.. Enter the required time limits under *Maximum minutes*:
+... If the employee must not work more than a set time per day, enable the check box at *Per Day* and enter the amount of minutes in the field next to this check box.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: These aren't steps, they are options, so should be a bulleted  list.

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+. To enter or change the list of contracts, click the *Contracts* tab.
++
+You can see the numbers of currently visible contracts in the top right part of the browser window, for example, `1-15 of 34`. You can use the *<* and *>* buttons to display other contracts in the list.
++
+You can enter any part of an contract name in the *Search* box to search for contracts.
++
+. Complete the following steps to add a new contract:
+.. Click *Add Contract*.
+.. Enter the name of the contract into the text field under *Name*.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: I'd say:
 Enter the contract name in the *Name* box. In any case, change into to to.

File: doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc [hunk 7]
```diff
@@ -1,20 +1,21 @@
 [id='er-employees-proc']
 = Entering the list of employees
 
-You must enter the list of all employees of the business and the skills they possess. The application rosters these employees to spots according to their skills.
+You must enter the list of all employees of the business, the skills they possess, and the contracts that apply to them. The application rosters these employees to spots according to their skills, observing the work time limits in the contracts.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc]: their skills and by observing...
or
their skills by obsering..

File: doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc [hunk 8]
```diff
@@ -1,20 +1,21 @@
...
+You can see the numbers of currently visible employees in the top right part of the browser window, for example, `1-15 of 34`. You can use the *<* and *>* buttons to display other employees in the list.
 +
-You can enter any part of an employee name in the *Search* box to search for employees. 
+You can enter any part of an employee name in the *Search* box to search for employees.
 +
 . Complete the following steps to add a new employee:
 .. Click *Add Employee*.
 .. Enter the name of the employee into the text field under *Employee name*.
 .. Optionally, select one or more skills from the drop-down list under *Skill set*.
+.. Select a contract from the drop-down list under *Contract*.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc]: Consider whether you really need to specify drop-down. IBMSG:
drop-down adj
Use only if you must specify the type of menu or list.

File: doc-content/enterprise-only/employee-rostering/optashift-ER-overview-con.adoc [hunk 9]
```diff
@@ -3,6 +3,6 @@
 
 The employee rostering starter application assigns employees to shifts on various positions in an organization. For example, you can use the application to distribute shifts in a hospital between nurses, guard duty shifts across a number of locations, or shifts on an assembly line between workers.
 
-Optimal employee rostering must take a number of variables into account. For example, different skills can be required for shifts in different positions. Also, some employees might be unavailable for some time slots or might prefer a particular time slot.
+Optimal employee rostering must take a number of variables into account. For example, different skills can be required for shifts in different positions. Also, some employees might be unavailable for some time slots or might prefer a particular time slot. Moreover, an employee can have a contract that limits the hours this employee can work in a single time period.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/optashift-ER-overview-con.adoc]: hours the employee can work

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc (hunk 4 above)
Comment (mramendi) [reply doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: In a user guide to an application, do we really give a prereq that the application must be started? Would you have any examples in existing RH docs? For example, in our Business Central docs, I see no prereqs in a procedure that you start in the main menu of BC, such as https://access.redhat.com/documentation/en-us/red_hat_process_automation_manager/7.3/html/designing_a_decision_service_using_guided_rule_templates/guided-rule-templates-create-proc

So I really don't see why I should add "prerequisites - you have an application..." to all my  procedures in Employee Rostering.

The restructure of the OpenShift docs was different. There, one needed to be in a very particular spot in the UI. Here, one just needs to have the app running - you can get into any tab from any place. And judging by our existing examples we don't make "app is started" an explicit prereq. I suggest not doing this here either, if only for the same of consistency.

File: doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc (hunk 8 above)
Comment (mramendi) [reply doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc]: I do think it is important to get the user to click and see it's drop-down. Just "the list" could cause a "what list? I don't see a list here" reaction.

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc (hunk 4 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: OK, but you at least need to say where to start from, as in the example you provide above.
To enter or change the list of contracts, click the Contracts tab. Where do I do this?

File: doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc (hunk 8 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/employee-rostering/er-employees-proc.adoc]: OK

File: doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc (hunk 4 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/employee-rostering/er-contracts-proc.adoc]: That's fine now.

File: doc-content/enterprise-only/employee-rostering/er-shiftroster-proc.adoc [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
 
-The Shift Roster is a table of all spots and all possible time spans. 
+The Shift Roster is a table of all spots and all possible time spans.
 
 If an employee must be present in a spot during a time span, a _shift_ must exist for this spot and this time span. If a spot requires several employees at the same time, you can create several shifts for the same spot and time span.
 
 Each shift is represented by a rectangle at the intersection of a spot (row) and time span (column).
 
-When new time is added to the draft period, the application copies the shifts (and default employees, if present) from the rotation pattern into this new part of the draft period. You can also manually add and edit shifts in the draft period. 
+When new time is added to the draft period, the application copies the shifts (and default employees, if present) from the rotation pattern into this new part of the draft period. You can also manually add and edit shifts in the draft period.
```
Comment (emmurphy1) [inline doc-content/enterprise-only/employee-rostering/er-shiftroster-proc.adoc]: Should this be 'when a new time'?

**Pull request merged**